### PR TITLE
codecs: merge 2 and 3 stubs

### PR DESCRIPTION
**Pull request by JelleZijlstra**
Related to #732.

There don't appear to be any major API differences in this module between Python 2 and 3. For reference, I made gists of the following:
- diff between previous py2 and py3 stubs: https://gist.github.com/JelleZijlstra/0aaeda5bb10cca3cf7fdc649ae2304ea
- diff between previous py2 stub and new stub from this PR: https://gist.github.com/JelleZijlstra/852a60f0fb1acf2da24b48a4f496851c
- diff between py3 stub and new 2and3 stub: https://gist.github.com/JelleZijlstra/49289b9d46e2f172070d811b4fbfb686

Some notes:
- I avoided making changes unrelated to merging, but did fix the return value of `StreamReaderWriter.__enter__` (which was missing in py3 and had a somewhat incorrect return value in py2).
- I made constructor bodies empty as much as possible (following the style of the py2 stub), in accordance with https://github.com/python/typeshed/pull/959.
- The `errors` argument to `EncodedFile` was marked as `Optional` in py2 but not py3. From reading the source in both versions, it doesn't appear to be Optional.

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
Actually, going to roll back `StreamReaderWriter.__enter__` to `BinaryIO` for now since this will probably break some existing code otherwise. I'll save cleaning that up for another PR.

File: stdlib/2and3/codecs.pyi [hunk 1]
```diff
@@ -15,13 +15,18 @@ from abc import abstractmethod
...
-# There *are* str->str and unicode->unicode encodings in the standard library.
-# And unlike python 3, they are in fairly widespread use.
+# There *are* bytes->bytes and str->str encodings in the standard library.
+# They are much more common in Python 2 than in Python 3.
+# Python 3.5 supposedly might change something there.
 
-_decoded = unicode
-_encoded = str
+if sys.version_info >= (3, 0):
+    _decoded = str
```
Comment (matthiaskramm) [inline stdlib/2and3/codecs.pyi]: Is this necessary? Could you just use "`Text`" for `_decoded`, and "`bytes`" for `_encoded`?

Comment (JelleZijlstra) [inline stdlib/2and3/codecs.pyi]: Thanks, pushed a commit removing the branching.

**Comment by gvanrossum** [comment]
Thanks -- I didn't review this very carefully (reviewing these kind of 2/3 merges is a pain) but if there's something wrong we'll find out soon enough.

**Pull request merged**